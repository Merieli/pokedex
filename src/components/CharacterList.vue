<script setup lang="ts">
import CharacterCard from '@/components/CharacterCard.vue';
import { useCharactersStore } from '@/infrastructure/store/characters';
import CardLoading from './CardLoading.vue';
const props = defineProps({});

const store = useCharactersStore();

store.getAllCharacters();
</script>

<template>
    <section class="character-list">
        <h4 class="character-list__qtd" data-list="qtd">100 Characters</h4>

        <CharacterCard
            name="Teste"
            :id="1"
            species="Human"
            image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            alt-image="nada por aqui"
            data-list="card"
        />
        <div class="h-10">{{ store.characters[0] }}</div>
        <CardLoading />
        <v-btn class="character-list__button" variant="tonal" data-list="button-load-more">Load more</v-btn>
    </section>
</template>

<style lang="postcss" scoped>
.character-list {
    @apply px-10 py-2 flex flex-col justify-center;

    &__qtd {
        @apply pb-6 font-semibold text-lg text-gray-500;
    }

    &__button {
        @apply rounded-2xl w-10/12 m-auto mt-6 bg-meri-mid text-white font-sans
        text-lg;
    }
}
</style>
