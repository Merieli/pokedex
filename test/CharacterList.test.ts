import CharacterList from '@components/CharacterList.vue';
import { mount, VueWrapper } from '@vue/test-utils';

describe('CharacterList.vue', () => {
    const wrapper: VueWrapper = mount(CharacterList);

    describe('Integração/Componente', () => {
        describe('👀 Renderização:', () => {
            test('Dado o componente Quando renderizado Então deve possuir html equivalente ao Snapshot gravado', () => {
                expect(wrapper.html()).toMatchSnapshot();
            });

            test('Dado uma lista de personagens Quando renderizada Então deve mostrar a quantidade de personagens total', () => {
                const qtd = wrapper.find('[data-list="qtd"]');

                expect(qtd.text()).toBe('100 Characters');
            });

            test('Dado uma lista de personagens Quando renderizada Então deve mostrar o card de personagens', () => {
                const qtd = wrapper.findAll('[data-list="card"]');

                expect(qtd).toHaveLength(1);
            });

            test('Dado uma lista de personagens Quando renderizada Então deve exibir um botão para carregar mais personagens', () => {
                const buttonLoadMore = wrapper.findAll('[data-list="button-load-more"]');

                expect(buttonLoadMore).toHaveLength(1);
            });
        });
        // describe('🧠 Comportamento:', () => {});
        // describe('🐕 Navegação:', () => {});
    });
});
