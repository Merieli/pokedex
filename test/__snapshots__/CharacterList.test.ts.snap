// Vitest Snapshot v1

exports[`CharacterList.vue > Integra√ß√£o/Componente > üëÄ Renderiza√ß√£o: > Dado o componente Quando renderizado Ent√£o deve possuir html equivalente ao Snapshot gravado 1`] = `
"<section class=\\"character-list\\" data-v-c3db22c7=\\"\\">
  <h4 class=\\"character-list__qtd\\" data-list=\\"qtd\\" data-v-c3db22c7=\\"\\">100 Characters</h4>
  <section class=\\"character-card\\" data-character=\\"card\\" style=\\"background-image: url(https://rickandmortyapi.com/api/character/avatar/1.jpeg);\\" data-list=\\"card\\" data-v-82d09848=\\"\\" data-v-c3db22c7=\\"\\">
    <p class=\\"character-card__id\\" data-character=\\"id\\" data-v-82d09848=\\"\\">#1</p>
    <h2 class=\\"character-card__name\\" data-character=\\"name\\" data-v-82d09848=\\"\\">Teste</h2>
    <div class=\\"character-card__tag\\" data-character=\\"species\\" data-v-82d09848=\\"\\">Human</div>
  </section>
  <div class=\\"h-10\\" data-v-c3db22c7=\\"\\"></div>
  <div class=\\"card-loading\\" data-v-73c53457=\\"\\" data-v-c3db22c7=\\"\\">
    <div class=\\"card-loading__content bg-gray-300 skeleton-loader\\" data-v-7049fcc4=\\"\\" data-v-73c53457=\\"\\">
      <div class=\\"skeleton-loader__shimmer bg-gray-300\\" data-v-7049fcc4=\\"\\"></div>
    </div>
    <div class=\\"card-loading__content bg-gray-300 skeleton-loader\\" data-v-7049fcc4=\\"\\" data-v-73c53457=\\"\\">
      <div class=\\"skeleton-loader__shimmer bg-gray-300\\" data-v-7049fcc4=\\"\\"></div>
    </div>
  </div>
  <v-btn class=\\"character-list__button\\" variant=\\"tonal\\" data-list=\\"button-load-more\\" data-v-c3db22c7=\\"\\">Load more</v-btn>
</section>"
`;
